data_science/specs/cohort_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rspec'
+require 'json'
+require_relative '../cohort.rb'
+
+RSpec.describe 'Cohort' do
+  let (:test_data) { JSON.parse(File.read('../test_data.json')) }
+  let (:cohort_a) { test_data.select { |v| v['cohort'] == 'A' } }
+
+  before :each do
+    @cohort = Cohort.new(name: 'A', data: cohort_a)
+  end
+
+  describe '#new' do
+    it 'returns a new Cohort object' do
+      expect(@cohort).to be_an_instance_of Cohort
+    end
+
+    it 'initializes with a string for a name' do
+      expect(Cohort.new.name).to be_a String
+    end
+
+    it 'initializes with an array of data' do
+      expect(Cohort.new.data).to be_a Array
+    end
+
+    it 'initializes with an empty data array' do
+      expect(Cohort.new.data).to be_empty
+    end
+
+    it 'initializes with an empty string for a name' do
+      expect(Cohort.new.name).to be_empty
+    end
+
+    it 'accepts a string as a name' do
+      expect(@cohort.name).to eq('A')
+    end
+
+    it 'accepts parsed JSON data' do
+      expect(@cohort.data).to_not be_empty
+    end
+
+    it 'returns an array of data' do
+      expect(@cohort.data).to be_a Array
+    end
+
+    it 'returns an array of Hashes' do
+      expect(@cohort.data.all? { |x| x.is_a? Hash }).to be true
```

Comment:
You can use the `all` RSpec matcher http://www.relishapp.com/rspec/rspec-expectations/v/3-3/docs/built-in-matchers/all-matcher:

```
expect(@cohort.data).to all( be_a(Hash) )
```